truffleruby/src/main/java/org/truffleruby/core/rubinius/IONodes.java

```diff
@@ -799,7 +801,9 @@ public int accept(DynamicObject io) {
             final int newFd;
 
             try {
-                newFd = ensureSuccessful(nativeSockets().accept(fd, memoryManager().newPointer(address), addressLength));
+                newFd = getContext().getThreadManager().runUntilResult(this,
+                        () -> ensureSuccessful(nativeSockets().accept(fd, memoryManager().newPointer(address), addressLength)),
+                        () -> nativeSockets().shutdown(fd, SHUT_RD));
```

Comment:
The man says:
```
 The  shutdown()  call  causes  all or part of a full-duplex connection on the socket associated
       with sockfd to be shut down.  If how is SHUT_RD, further receptions will be disallowed. 
```
But indeed we should verify the actual behavior if `accept` is called again.